refinery/core/views.py

```diff
@@ -1434,18 +1434,37 @@ class SiteProfileAPIView(APIView):
     http_method_names = ["get", "patch"]
 
     def get(self, request):
-        try:
-            site_profile = SiteProfile.objects.get(
-                site=get_current_site(request)
-            )
-        except SiteProfile.DoesNotExist as e:
-            logger.error("Site profile for the current site does not exist.")
-            return HttpResponseNotFound(e)
-        except SiteProfile.MultipleObjectsReturned:
-            logger.error("Multiple site profiles for current site error.")
-            return HttpResponseServerError(e)
-
-        serializer = SiteProfileSerializer(site_profile)
+        """
+        Returns the current profile ('current' param) or a list of profiles
+        :param request: API request
+        :return: serialized site profile(s)
+        """
+        if request.query_params.get('current_site'):
```

Comment:
1) Default to return a list unless current_site is true. If current_site is false, missing, or anything else just return the list.
2) all_site_profiles_serialized is unnecessary